hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionState.java

```diff
@@ -69,28 +70,30 @@ public SICompactionState(TxnSupplier transactionStore, int activeTransactionCach
      * @param rawList - the input of key values to process
      * @param results - the output key values
      */
-    public void mutate(List<Cell> rawList, List<TxnView> txns, List<Cell> results, boolean purgeDeletedRows) throws IOException {
+    public void mutate(List<Cell> rawList, List<TxnView> txns, List<Cell> results, boolean forcePurgingDeletedRows, boolean keepTombstones) throws IOException {
         dataToReturn.clear();
+        long lowWatermarkTransaction = TransactionsWatcher.INSTANCE.getLowWatermarkTransaction();
         long maxTombstone = 0;
         Iterator<TxnView> it = txns.iterator();
         for (Cell aRawList : rawList) {
             TxnView txn = it.next();
             long t = mutate(aRawList, txn);
-            if (t > maxTombstone) {
+            if (maxTombstone < t && t < lowWatermarkTransaction)
```

Comment:
Here I think we are processing all tombstones, not just only the committed ones. Tombstones from Active transactions should be ignored due to the lowWatermarkTransaction check, but I think we'd be removing data based on rolledback tombstones.

Reply:
Good catch. That should be addressed in 807e48d